File: arreglos 27-08-2025/Arreglos.cs

using System;

public class Program
{
    public static void Main(string[] args)
    {
        // 1. Declaración e inicialización
        // Declarar e inicializar un arreglo de enteros con valores.
        int[] numeros = { 10, 20, 30, 40, 50 };
        Console.WriteLine($"Arreglo inicial: [{string.Join(", ", numeros)}]");

        // 2. Acceder a elementos
        // Los índices comienzan en 0.
        int primerNumero = numeros[0]; // Accede al 10
        Console.WriteLine($"El primer número es: {primerNumero}");

        // 3. Modificar elementos
        numeros[2] = 35; // Cambia el tercer elemento (30) a 35.
        Console.WriteLine($"Arreglo modificado: [{string.Join(", ", numeros)}]");

        // 4. Longitud del arreglo
        // La propiedad Length contiene el número de elementos.
        Console.WriteLine($"La longitud del arreglo es: {numeros.Length}");

        // 5. Iterar sobre un arreglo (usando un bucle foreach)
        Console.WriteLine("Iterando sobre el arreglo:");
        foreach (int numero in numeros)
        {
            Console.WriteLine($"- {numero}");
        }

    }
}

File: arreglos 27-08-2025/Arreglos.java

// En Java, los arreglos son objetos que contienen un número fijo de valores de un solo tipo.
// La longitud de un arreglo se establece cuando se crea y no puede cambiar.
// Para colecciones dinámicas, se suele usar ArrayList.

import java.util.Arrays;

public class Arreglos {
    public static void main(String[] args) {
        // 1. Declaración e inicialización
        // Declarar un arreglo de enteros
        int[] numeros;

        // Inicializar el arreglo con un tamaño de 5 (los valores por defecto son 0)
        numeros = new int[5];

        // Asignamos valores a nuestro arreglo de números
        numeros[0] = 100;
        numeros[1] = 200;
        System.out.println("Arreglo de números: " + Arrays.toString(numeros));

        // Declarar e inicializar con valores en una sola línea
        String[] frutas = {"Manzana", "Banana", "Naranja"};
        System.out.println("Arreglo inicial de frutas: " + Arrays.toString(frutas));

        // 2. Acceder a elementos
        // Los índices comienzan en 0
        String primeraFruta = frutas[0]; // Accede a "Manzana"
        System.out.println("La primera fruta es: " + primeraFruta);

        // 3. Modificar elementos
        frutas[1] = "Uva"; // Cambia "Banana" por "Uva"
        System.out.println("Arreglo modificado: " + Arrays.toString(frutas));

        // 4. Longitud del arreglo
        // .length es una propiedad, no un método
        System.out.println("La longitud del arreglo es: " + frutas.length);

        // 5. Iterar sobre un arreglo (usando un bucle "for-each")
        System.out.println("Iterando sobre el arreglo:");
        for (String fruta : frutas) {
            System.out.println("- " + fruta);
        }
    }
}
